omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PatientIdentifierResource1_8.java

```diff
@@ -89,11 +90,12 @@ public DelegatingResourceDescription getCreatableProperties() {
 	public void setIdentifierType(PatientIdentifier instance, PatientIdentifierType identifierType) {
 		String uuid = identifierType.getUuid();
 		String name = identifierType.getName();
-		if (name != null && !name.isEmpty()) {
-			instance.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByName(name));
-		} else if (uuid != null && !uuid.isEmpty()) {
+		if (!StringUtils.isEmpty(uuid)) {
 			instance.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
 		}
+		if (instance.getIdentifierType() == null && !StringUtils.isEmpty(name)) {
```

Comment:
Ah sure.
Anyway I'll close this since I need to switch branches on our fork. We'll follow up on the new PR.
Thanks for your analysis.